stl/inc/ranges

```diff
@@ -7656,6 +7662,401 @@ namespace ranges {
+
+        template <bool _IsConst>
+        using _Zentinel = sentinel_t<_Maybe_const<_IsConst, _Inner_view>>;
+
+        template <bool _IsConst>
+        struct _Category_base {};
+
+        template <bool _IsConst>
+            requires forward_range<_Maybe_const<_IsConst, _Inner_view>>
+        struct _Category_base<_IsConst> {
+            using iterator_category = decltype([]() noexcept {
+                using _Invoke_t = invoke_result_t<_Maybe_const<_IsConst, _Func>&,
+                    range_reference_t<_Maybe_const<_IsConst, _ViewTypes>>...>;
+
+                if constexpr (!is_reference_v<_Invoke_t>) {
+                    return input_iterator_tag{};
+                } else {
+                    constexpr auto _Check_tag_type_closure = []<typename _Tag_type>() noexcept {
+                        // clang-format off
+                        return (derived_from<typename
+                            iterator_traits<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>::iterator_category,
+                            _Tag_type>
+                        && ...);
+                        // clang-format on
+                    };
+
+                    if constexpr (_Check_tag_type_closure.template operator()<random_access_iterator_tag>()) {
+                        return random_access_iterator_tag{};
+                    } else if constexpr (_Check_tag_type_closure.template operator()<bidirectional_iterator_tag>()) {
+                        return bidirectional_iterator_tag{};
+                    } else if constexpr (_Check_tag_type_closure.template operator()<forward_iterator_tag>()) {
+                        return forward_iterator_tag{};
+                    } else {
+                        return input_iterator_tag{};
+                    }
+                }
+            }());
+        };
+
+        template <bool _IsConst>
+        class _Iterator : public _Category_base<_IsConst> {
+        private:
+            friend zip_transform_view;
+
+            using _Parent_t = _Maybe_const<_IsConst, zip_transform_view>;
+            using _Base_t   = _Maybe_const<_IsConst, _Inner_view>;
+
+            _Parent_t* _Parent = nullptr;
+            /* [[no_unique_address]] */ _Ziperator<_IsConst> _Inner;
+
+            constexpr _Iterator(_Parent_t& _Parent_, _Ziperator<_IsConst> _Inner_) noexcept(
+                is_nothrow_move_constructible_v<_Ziperator<_IsConst>>) // strengthened
+                : _Parent(_STD addressof(_Parent_)), _Inner(_STD move(_Inner_)) {}
+
+        public:
+            using iterator_concept = typename _Ziperator<_IsConst>::iterator_concept;
+            using value_type       = remove_cvref_t<invoke_result_t<_Maybe_const<_IsConst, _Func>&,
+                range_reference_t<_Maybe_const<_IsConst, _ViewTypes>>...>>;
+            using difference_type  = range_difference_t<_Base_t>;
+
+            _Iterator() noexcept(is_nothrow_default_constructible_v<_Ziperator<_IsConst>>) = default;

```

Comment:
While this is correct, I think that simply omitting the `noexcept` specification would be reasonable, as it exactly matches what the compiler will generate (and the Standard simply depicts `iterator () = default;`).

Occurs below for `_Sentinel()` and `zip_transform_view()`.